doom-modeline-segments.el

```diff
@@ -1960,18 +1960,24 @@ mouse-3: Fetch notifications"
 ;;
 
 ;; Highlight the mode-line while debugging.
-(defvar-local doom-modeline--debug-cookie nil)
+(defvar doom-modeline--debug-cookie nil)
```

Comment:
This need to be `defvar-local` or else you're overwriting the value with every buffer

Reply:
Yeah, the `when` was in the wrong position, thanks!